public/app/features/dashboard/components/Inspector/PanelInspector.tsx

```diff
@@ -21,16 +22,26 @@ export interface ConnectedProps {
   plugin?: PanelPlugin | null;
 }
 
-export type Props = OwnProps & ConnectedProps;
+interface DispatchProps {
+  updateLocation: typeof updateLocation;
+}
+
+export type Props = OwnProps & ConnectedProps & DispatchProps;
 
-const PanelInspectorUnconnected: React.FC<Props> = ({ panel, dashboard, defaultTab, plugin }) => {
+const PanelInspectorUnconnected: React.FC<Props> = ({ panel, dashboard, defaultTab, plugin, updateLocation }) => {
   const [dataOptions, setDataOptions] = useState<GetDataOptions>({
     withTransforms: false,
     withFieldConfig: false,
   });
-  const [lastResult, isLoading, error] = usePanelLatestData(panel, dataOptions);
-  const metaDs = useDatasourceMetadata(lastResult);
+  const { data, isLoading, error } = usePanelLatestData(panel, dataOptions);
+  const metaDs = useDatasourceMetadata(data);
   const tabs = useInspectTabs(plugin, dashboard, error, metaDs);
+  const onClose = useCallback(() => {
+    updateLocation({
+      query: { inspect: null, inspectTab: null },
+      partial: true,
+    });
+  }, [updateLocation]);
```

Comment:
Couldn't this be solved with `useDispatch` instead or are we missing the context here?

Reply:
hmmm, never used useDispatch. Can take a look at it

Reply:
worked nicely!